gpbft/gpbft.go

```diff
@@ -373,8 +374,14 @@ func (i *instance) receiveOne(msg *GMessage) (bool, error) {
 	msgRound := i.getRound(msg.Vote.Round)
 	switch msg.Vote.Step {
 	case QUALITY_PHASE:
-		// Receive each prefix of the proposal independently.
+		// Receive each prefix of the proposal independently, which is accepted at any
+		// round/phase.
 		i.quality.ReceiveEachPrefix(msg.Sender, msg.Vote.Value)
+		// If the instance has surpassed QUALITY phase, update the candidates based
+		// on possible quorum of input prefixes.
+		if i.phase != QUALITY_PHASE {
+			return true, i.tryQuality()
```

Comment:
I'm not a fan:

1. Here, we check "if not quality, call try quality".
2. Then, in "try quality", we have a completely separate code path for "not quality".

That should just be an entirely different function.

Reply:
That makes sense